### DISCUSSION TITLE
bad route when using encore_entry_link_tags('app') or encore_entry_script_tags('app')

### DISCUSSION THREAD

**Discussion by Vincezo26740** [symfony/symfony#47942] [Q&A]
Hi, i've got a little problem with symfony encore web-pack.
I had done some Javascript line code and want to put it in my twig.
i install encore web-pack bundle. when i use the localhost i've got my first route ok.
My second route in my controller, don't work. I install finally apache-pack and it is ok for links but link to css and js are wrong.

my controller : 
```php
    #[Route('/', name: 'app_bateau')]
    public function index(BateauRepository $bateauRepository): Response
    {
        return $this->render('bateau/index.html.twig', [
            'bateaux' => $bateauRepository->findAll()
        ]);
    }

    #[Route('/bateau_create', name: 'bateau_create')]
    public function create(Request $request): Response
    {
        $form = $this->createForm(BateauType::class)->handleRequest($request);
        return $this->render('bateau/create.html.twig', [
            'form' => $form->createView()
        ]);
    }
```

my htaccess : 

```
DirectoryIndex index.php

<IfModule mod_negotiation.c>
    Options -MultiViews
</IfModule>

<IfModule mod_rewrite.c>
    RewriteEngine On

    RewriteCond %{REQUEST_URI}::$0 ^(/.+)/(.*)::\2$
    RewriteRule .* - [E=BASE:%1]

    RewriteCond %{HTTP:Authorization} .+
    RewriteRule ^ - [E=HTTP_AUTHORIZATION:%0]

   RewriteCond %{ENV:REDIRECT_STATUS} =""
    RewriteRule ^index\.php(?:/(.*)|$) %{ENV:BASE}/$1 [R=301,L]

    RewriteCond %{REQUEST_FILENAME} !-f
    RewriteRule ^ %{ENV:BASE}/index.php [L]
</IfModule>

<IfModule !mod_rewrite.c>
    <IfModule mod_alias.c>

        RedirectMatch 307 ^/$ /index.php/

    </IfModule>
</IfModule>
```

my html code view : 
```html
<head>
		<meta charset="UTF-8">
		<title>
			Welcome!
					</title>
							<link rel="stylesheet" href="/build/app.css">
		
					<script src="/build/runtime.js" defer></script><script src="/build/vendors-node_modules_core-js_modules_es_array_for-each_js-node_modules_core-js_modules_es_obj-651634.js" defer></script><script src="/build/app.js" defer></script>
			</head>
```

Thanks for your help

**Comment by stof** [comment]
What do you mean by "they are wrong" ?

**Reply by Vincezo26740** [reply]
Hi. Thanks for your answer.
Well. When i use "asset" commande the Links don't work. I checker Links and they are " localhost/build/app.js" and when i check with F12 on console mode, the message say it don't find document. I suppose i've Got wrong redirection in .htaccess or in vhost.conf.
I try to unload .htaccess but redirection don't work.

**Reply by Vincezo26740** [reply]
I write hard link and don't work like i would. I've got a little js to put in my code so i write it directly and it works.
to resume : 
- install webpack encore
- doing my code
- doing run watch ( build is correctly, new files and directory )
- try with firefox, 1st page ok, link to go to 2nd page don't work
- install apache-pack
- retry in firefox, all link to redirect ok but not link for Js or Css ( error don't find files )
- try with symphony server, all work ( links and Js )
- try to leave .htaccess, links and Js don't work.
- No more idea so go here ;-)

**Reply by Vincezo26740** [reply]
i'm using Wamp64, Vscode, commander, sqlite, windows 11

**Reply by Vincezo26740** [reply]
Just a question comes, Why it works when i use symfony server ???
